src/components/ui/avatar-frame.tsx

```diff
@@ -0,0 +1,46 @@
+import { useState } from 'react';
+
+interface AvatarFrameProps {
+  src: string;
+  alt: string;
+  width?: number;
+  showGrid?: boolean;
+  height?: number;
+}
+
+export default function AvatarFrame({
+  src,
+  alt,
+  width = 50, // Default values
+  height = 50,
+  showGrid = true,
+}: AvatarFrameProps) {
+  const [isLoaded, setIsLoaded] = useState(false);
+
+  return (
+    <div className="relative" style={{ width: `${width}px`, height: `${height}px` }}>
+      {showGrid && (
+        <div
+          className="absolute inset-0 z-0 rounded-full"
+          style={{
```

Comment:
Try to rely on Tailwind CSS classes or custom Tailwind styles as much as possible. Using inline styles for simple logic will only increase CSS specificity, making it harder to manage when we overwrite it later.